File: src/main/java/net/atos/zac/app/policy/converter/RESTRechtenConverter.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.app.policy.converter;

import net.atos.zac.app.policy.model.RESTDocumentRechten;
import net.atos.zac.app.policy.model.RESTOverigeRechten;
import net.atos.zac.app.policy.model.RESTTaakRechten;
import net.atos.zac.app.policy.model.RESTWerklijstRechten;
import net.atos.zac.app.policy.model.RESTZaakRechten;
import net.atos.zac.policy.output.DocumentRechten;
import net.atos.zac.policy.output.OverigeRechten;
import net.atos.zac.policy.output.TaakRechten;
import net.atos.zac.policy.output.WerklijstRechten;
import net.atos.zac.policy.output.ZaakRechten;

public class RESTRechtenConverter {

    public RESTDocumentRechten convert(final DocumentRechten documentRechten) {
        final RESTDocumentRechten restDocumentRechten = new RESTDocumentRechten();
        restDocumentRechten.lezen = documentRechten.lezen();
        restDocumentRechten.wijzigen = documentRechten.wijzigen();
        restDocumentRechten.ontgrendelen = documentRechten.ontgrendelen();
        restDocumentRechten.vergrendelen = documentRechten.vergrendelen();
        restDocumentRechten.verwijderen = documentRechten.verwijderen();
        restDocumentRechten.ondertekenen = documentRechten.ondertekenen();
        return restDocumentRechten;
    }

    public RESTTaakRechten convert(final TaakRechten taakRechten) {
        final RESTTaakRechten restTaakRechten = new RESTTaakRechten();
        restTaakRechten.lezen = taakRechten.lezen();
        restTaakRechten.wijzigen = taakRechten.wijzigen();
        restTaakRechten.toekennen = taakRechten.toekennen();
        restTaakRechten.toevoegenDocument = taakRechten.toevoegenDocument();
        return restTaakRechten;
    }

    public RESTZaakRechten convert(final ZaakRechten zaakRechten) {
        final RESTZaakRechten restZaakRechten = new RESTZaakRechten();
        restZaakRechten.lezen = zaakRechten.lezen();
        restZaakRechten.wijzigen = zaakRechten.wijzigen();
        restZaakRechten.toekennen = zaakRechten.toekennen();
        restZaakRechten.behandelen = zaakRechten.behandelen();
        restZaakRechten.afbreken = zaakRechten.afbreken();
        restZaakRechten.heropenen = zaakRechten.heropenen();
        restZaakRechten.wijzigenDoorlooptijd = zaakRechten.wijzigenDoorlooptijd();
        restZaakRechten.bekijkenZaakdata = zaakRechten.bekijkenZaakdata();
        return restZaakRechten;
    }

    public RESTWerklijstRechten convert(final WerklijstRechten werklijstrechten) {
        final RESTWerklijstRechten restWerklijstRechten = new RESTWerklijstRechten();
        restWerklijstRechten.inbox = werklijstrechten.inbox();
        restWerklijstRechten.ontkoppeldeDocumentenVerwijderen = werklijstrechten.ontkoppeldeDocumentenVerwijderen();
        restWerklijstRechten.inboxProductaanvragenVerwijderen = werklijstrechten.inboxProductaanvragenVerwijderen();
        restWerklijstRechten.zakenTaken = werklijstrechten.zakenTaken();
        restWerklijstRechten.zakenTakenVerdelen = werklijstrechten.zakenTakenVerdelen();
        restWerklijstRechten.zakenTakenExporteren = werklijstrechten.zakenTakenExporteren();
        return restWerklijstRechten;
    }

    public RESTOverigeRechten convert(final OverigeRechten overigeRechten) {
        final RESTOverigeRechten restOverigeRechten = new RESTOverigeRechten();
        restOverigeRechten.startenZaak = overigeRechten.startenZaak();
        restOverigeRechten.beheren = overigeRechten.beheren();
        restOverigeRechten.zoeken = overigeRechten.zoeken();
        return restOverigeRechten;
    }
}

File: src/main/java/net/atos/zac/app/policy/converter/RestRechtenConverter.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.policy.converter

import net.atos.zac.app.policy.model.RestDocumentRechten
import net.atos.zac.app.policy.model.RestOverigeRechten
import net.atos.zac.app.policy.model.RestTaakRechten
import net.atos.zac.app.policy.model.RestWerklijstRechten
import net.atos.zac.app.policy.model.RestZaakRechten
import nl.info.zac.policy.output.DocumentRechten
import nl.info.zac.policy.output.OverigeRechten
import nl.info.zac.policy.output.TaakRechten
import nl.info.zac.policy.output.WerklijstRechten
import nl.info.zac.policy.output.ZaakRechten

fun DocumentRechten.toRestDocumentRechten() = RestDocumentRechten(
    lezen = this.lezen,
    wijzigen = this.wijzigen,
    ontgrendelen = this.ontgrendelen,
    vergrendelen = this.vergrendelen,
    verwijderen = this.verwijderen,
    ondertekenen = this.ondertekenen,
    toevoegenNieuweVersie = this.toevoegenNieuweVersie
)

fun TaakRechten.toRestTaakRechten() = RestTaakRechten(
    lezen = this.lezen,
    wijzigen = this.wijzigen,
    toekennen = this.toekennen,
    toevoegenDocument = this.toevoegenDocument
)

fun ZaakRechten.toRestZaakRechten() = RestZaakRechten(
    lezen = this.lezen,
    wijzigen = this.wijzigen,
    toekennen = this.toekennen,
    behandelen = this.behandelen,
    afbreken = this.afbreken,
    heropenen = this.heropenen,
    wijzigenDoorlooptijd = this.wijzigenDoorlooptijd,
    bekijkenZaakdata = this.bekijkenZaakdata,
    versturenOntvangstbevestiging = this.versturenOntvangstbevestiging,
    toevoegenBagObject = this.toevoegenBagObject,
    toevoegenBetrokkeneBedrijf = this.toevoegenBetrokkeneBedrijf,
    toevoegenBetrokkenePersoon = this.toevoegenBetrokkenePersoon,
    toevoegenInitiatorBedrijf = this.toevoegenInitiatorBedrijf,
    toevoegenInitiatorPersoon = this.toevoegenInitiatorPersoon,
    verwijderenBetrokkene = this.verwijderenBetrokkene,
    verwijderenInitiator = this.verwijderenInitiator,
    creeerenDocument = this.creeerenDocument,
    versturenEmail = this.versturenEmail,
    wijzigenLocatie = this.wijzigenLocatie
)

fun WerklijstRechten.toRestWerklijstRechten() = RestWerklijstRechten(
    inbox = this.inbox,
    ontkoppeldeDocumentenVerwijderen = this.ontkoppeldeDocumentenVerwijderen,
    inboxProductaanvragenVerwijderen = this.inboxProductaanvragenVerwijderen,
    zakenTaken = this.zakenTaken,
    zakenTakenVerdelen = this.zakenTakenVerdelen,
    zakenTakenExporteren = this.zakenTakenExporteren
)

fun OverigeRechten.toRestOverigeRechten() = RestOverigeRechten(
    startenZaak = this.startenZaak,
    beheren = this.beheren,
    zoeken = this.zoeken
)
